client/internal/internal.go

```diff
@@ -0,0 +1,71 @@
+package internal
+import (
+    "github.com/derekparker/delve/proctl"
+    "github.com/derekparker/delve/goreadline"
+    "fmt"
+    "io"
+    "strings"
+    "os"
+    sys "golang.org/x/sys/unix"
+)
+const HistoryFile string = ".dbg_history"
+
+
+func HandleExit(dbp *proctl.DebuggedProcess, status int, shouldConfirm bool) {
+    errno := goreadline.WriteHistoryToFile(HistoryFile)
```

Comment:
My later comment may invalidate this - but we shouldn't be doing anything with readline in the shared package. It should only be the `cli` client that needs it.